jme3-examples/src/main/java/jme3test/terrain/PBRTerrainAdvancedTest.java

```diff
@@ -0,0 +1,395 @@
+package jme3test.terrain;
+
+
+
+import com.jme3.app.SimpleApplication;
+import com.jme3.asset.TextureKey;
+import com.jme3.bounding.BoundingBox;
+import com.jme3.bounding.BoundingSphere;
+import com.jme3.font.BitmapText;
+import com.jme3.input.KeyInput;
+import com.jme3.input.controls.ActionListener;
+import com.jme3.input.controls.KeyTrigger;
+import com.jme3.light.AmbientLight;
+import com.jme3.light.DirectionalLight;
+import com.jme3.light.LightProbe;
+import com.jme3.material.Material;
+import com.jme3.math.ColorRGBA;
+import com.jme3.math.Vector3f;
+import com.jme3.scene.Node;
+import com.jme3.shader.VarType;
+import com.jme3.system.AppSettings;
+import com.jme3.terrain.geomipmap.TerrainQuad;
+import com.jme3.terrain.heightmap.AbstractHeightMap;
+import com.jme3.terrain.heightmap.ImageBasedHeightMap;
+import com.jme3.texture.Image;
+import com.jme3.texture.Texture;
+import com.jme3.texture.Texture.WrapMode;
+import com.jme3.texture.TextureArray;
+import java.util.ArrayList;
+import java.util.List;
+
+
+// Contains assets from CC0Textures.com, licensed under CC0 1.0 Universal.
+
+public class PBRTerrainAdvancedTest extends SimpleApplication {    
+    
+    
+    //similar to PRBTerrainTest, but this one uses texture arrays to allow for more texture slots. beware that GPUs for older devices and mobile devices may be less likely to support texture arrays
+        
+    //results look better with anti-aliasing, especially at far distances. 
+    
+    private TerrainQuad terrain;
+    private Material matTerrain;
+    private boolean triPlanar = false;
+    private float dirtScale = 24;
+    private float darkRockScale = 24;
+    private float snowScale = 64;
+    private float tileRoadScale = 64;
+    private float grassScale = 24;
+    private float marbleScale = 64;
+    private float gravelScale = 64;
+    
+    private AmbientLight ambientLight;
+    private DirectionalLight directionalLight;
+    private boolean isNight = false;
+    
+    private BitmapText keybindingsText;
+
+    public static void main(String[] args) {
+        PBRTerrainAdvancedTest app = new PBRTerrainAdvancedTest();
+        AppSettings s = new AppSettings(true);
+        s.put("FrameRate", 140);
```

Comment:
Why does this app set the frame rate? Shouldn't it synchronize with the graphics adapter/monitor?

Reply:
Calling that line sets 140 fps as the max, but it can still fall to any number below 140 fps.

I can remove it if you'd like, but my GPU makes weird noises when apps do not cap the framerate, so anytime I run a test case I add this line of code so my GPU isn't running at 9000 fps.

Reply:
When you run the test, tick the box that says "frame sync" or v sync or whatever.  Setting the frame rate as you are will cause the application to sleep periodically.

Reply:
Okay I will remove it from both test cases.

For reference for other projects: I see that settings VSync true will cap the app at 60fps, is there any way to increase this without setting the framerate as I've previously done?

Reply:
Vsync is locking refresh to the display's refresh rate.  It prevents tearing and prevents drawing frames that are just going to be thrown away anyway.  Also, at least in my day, the app can begin drawing the next frame already while the driver waits for the vsync.

I'm not sure why you would want to "increase it" unless you think "FPS" is a more useful metric than "how long did my frame take to render"... but "how long did my frame take to render" is a Waaaaaaay more useful metric and already provided by the built in profiling.

Reply:
I personally have no problem with my apps being capped at 60fps, however I know some younger kids who are into mainstream games nowadays like to brag about how their computer runs a game at 120 fps while their friends only run at 80, so it is really not important, but that is the only reason I asked. 
There are undoubtedly many kids who think a game is bad for running at 60 fps when they play fortnite at 120 fps, however irrational that thought may be.

Reply:
They are free to uncheck the box.

Making "vsync run faster" doesn't make any sense, though.  It's like "how can I make a day longer than a day?"